docker/reqgenerator.py

```diff
@@ -8,16 +8,23 @@
 
 
 def build_docker_deps(image_name: str, imagedir: str) -> None:
-    print("Fetch the newest manifest for", image_name)
+    print("Get the directory of packages installed by `pip`")
+    cmd = (
+        rf"docker run --network=host --rm --entrypoint 'python3' {image_name} -c "
+        """'import sys; print([p for p in sys.path if "/usr/local/lib/python" in p][0])'"""
+    )
+    dist_dir = str(subprocess.check_output(cmd, shell=True, encoding="utf-8").strip())
     pip_cmd = (
-        "pip install pipdeptree 2>/dev/null 1>/dev/null && pipdeptree --freeze "
-        "--warn silence --exclude pipdeptree"
+        "pip install pipdeptree 2>/dev/null 1>/dev/null && "
+        f"pipdeptree --freeze --warn silence --exclude pipdeptree --path {dist_dir}"
```

Comment:
Is it worth adding a comment explaining why want to freeze only `/usr/local/lib/python`?

Reply:
It's in line 11, ```print("Get the directory of packages installed by `pip`")```

Freezing distributive-installed packages shuts our leg right to the brain. Some of them, e.g. PyGObject, are installed even without proper dependencies. It's very much broken.

So, we freeze what we actually install.